### ISSUE TITLE
Error: write after end

### ISSUE THREAD

**Issue by franalt** [eleith/emailjs#205]
Hello everyone.

I'am getting this error, and I can´t find anywhere what it means. I have a node app that sends an array of emails and at some point this error ocurrs. Does anyone know what does it mean?

```
Error: write after end
    at writeAfterEnd (_stream_writable.js:236:12)
    at TLSSocket.Writable.write (_stream_writable.js:287:5)
    at TLSSocket.Socket.write (net.js:697:40)
    at SMTP.send (/app/node_modules/emailjs/smtp/smtp.js:186:17)
    at SMTP.command (/app/node_modules/emailjs/smtp/smtp.js:212:10)
    at SMTP.rset (/app/node_modules/emailjs/smtp/smtp.js:364:10)
    at check (/app/node_modules/emailjs/smtp/client.js:156:23)
    at caller (/app/node_modules/emailjs/smtp/smtp.js:48:14)
    at response (/app/node_modules/emailjs/smtp/smtp.js:198:9)
    at caller (/app/node_modules/emailjs/smtp/smtp.js:48:14)
```

This is my code:

```
server.send({
	text:    send_array[idx].client_email, 
	from:    sender_email, 
	to:      send_array[idx].client_to,
	cc:      '',
	bcc:     "server@adms.com.ar",
	subject: send_array[idx].client_subject + ' SID:' + idx,
	attachment: attachments
}, function(err, message) { 
        //send the next email if there is any
});	 
```

Thanks!

**Comment by ysilvela** [comment]
Me too. This error appear a few days ago.

**Comment by eleith** [comment]
what version of node are you running?

**Comment by eleith** [comment]
can you turn on debugging? (uncomment this line: https://github.com/eleith/emailjs/blob/master/smtp/client.js#L9)

it looks like for some reason you are sending a `RSET` because the SMTP server hit an error and likely closed, but the closing got missed by this library somehow.

**Comment by eleith** [comment]
so somewhere along the lines, `close(true)` was called. any chance you can do some more debugging to determine which one was called?

for example, was it this one? https://github.com/eleith/emailjs/blob/master/smtp/smtp.js#L164

**Comment by tjrhodes** [comment]
Getting masses of these errors, out of 170 mails, only 3 just got sent, rest were all "Error: connection encountered an error (write after end)" or "Error: write after end". Node 8.11.3
, getting them to non gmail and gmail smtp servers.

**Comment by zackschuster** [comment]
can anyone here verify these issues are still occurring?

**Comment by zackschuster** [comment]
this should be fixed by 81e78bfb4d44776528ff733c967013affa6fed27

will re-post when that has been released.

**Comment by zackschuster** [comment]
`3.3.0` has been release with this fix 😄
